homeassistant/components/swisscom/device_tracker.py

```diff
@@ -77,31 +80,146 @@ def _update_info(self):
         self.last_results = active_clients
         return True
 
-    def get_swisscom_data(self):
-        """Retrieve data from Swisscom and return parsed result."""
-        url = f"http://{self.host}/ws"
-        headers = {"Content-Type": "application/x-sah-ws-4-call+json"}
-        data = """
-        {"service":"Devices", "method":"get",
-        "parameters":{"expression":"lan and not self"}}"""
-
-        devices = {}
-
+    def swisscom_post_request(self, session, headers, data):
+        # Send a POST request to /ws endpoint of the router.
+        # Returns None on failure or the response object
+        url = f"https://{self.host}/ws"
+        _LOGGER.debug(
+            f"Send request to {url}:\nheaders={headers} \ndata={data} \ncookies={session.cookies}"
+        )
```

Comment:
This whole function should be moved to an external library, as per https://developers.home-assistant.io/docs/development_checklist/

> All communication to external devices or services must be wrapped in an external Python library hosted on [pypi](https://pypi.python.org/pypi).